src/payment/strategies/affirm/affirm-payment-strategy.spec.ts

```diff
@@ -0,0 +1,261 @@
+            .mockReturnValue(submitPaymentAction);
+
+        jest.spyOn(store.getState().paymentMethods, 'getPaymentMethod')
+            .mockReturnValue(paymentMethod);
+
+        jest.spyOn(paymentMethodActionCreator, 'loadPaymentMethod')
+            .mockReturnValue(loadPaymentMethodAction);
+
+        jest.spyOn(scriptLoader, 'load').mockReturnValue(Promise.resolve(affirm));
+
+        payload = merge({}, getOrderRequestBody(), {
+            payment: {
+                methodId: paymentMethod.id,
+                gatewayId: paymentMethod.gateway,
+            },
+        });
+    });
+
+    describe('#initialize()', () => {
+        it('throws error if client token is missing', async () => {
+            paymentMethod.clientToken = '';
+            try {
+
+                await strategy.initialize({ methodId: paymentMethod.id });
+            } catch (error) {
+                expect(error).toBeInstanceOf(MissingDataError);
+            }
+        });
+
+        it('loads affirm script from snippet', async () => {
+            await strategy.initialize({ methodId: paymentMethod.id });
+            expect(scriptLoader.load).toBeCalledWith(paymentMethod.clientToken, false);
+        });
+    });
+
+    describe('#execute()', () => {
+
+        beforeEach(async () => {
+            await strategy.initialize({ methodId: paymentMethod.id, gatewayId: paymentMethod.gateway });
+
+            jest.spyOn(store, 'dispatch').mockReturnValue(Promise.resolve());
+            jest.spyOn(affirm.checkout, 'open').mockImplementation(({ onSuccess }) => {
+                onSuccess({
+                    checkout_token: '1234',
+                    created: '1234',
+                });
+            });
+            jest.spyOn(store.getState().paymentMethods, 'getPaymentMethod').mockReturnValue(paymentMethod);
+
+        });
+
+        it('creates order, checkout and payment', async () => {
+            const options = { methodId: 'affirm', gatewayId: undefined };
+
+            await strategy.execute(payload, options);
+
+            expect(store.dispatch).toHaveBeenCalledWith(submitOrderAction);
+
+            expect(orderActionCreator.submitOrder).toHaveBeenCalledWith({ useStoreCredit: false }, options);
+            expect(affirm.checkout).toHaveBeenCalled();
+            expect(affirm.checkout.open).toHaveBeenCalled();
+            expect(affirm.ui.error.on).toHaveBeenCalled();
+            expect(store.dispatch).toHaveBeenCalledWith(submitPaymentAction);
+            expect(paymentActionCreator.submitPayment).toBeCalledWith({
+                methodId: paymentMethod.id,
+                paymentData: { nonce: '1234' },
+            });
+
+        });
+
+        it('returns error on affirm if users cancel flow', async () => {
+            jest.spyOn(affirm.checkout, 'open').mockImplementation(({ onFail }) => {
+                onFail();
+            });
+            try {
+                await strategy.execute(payload);
+            } catch (error) {
```

Comment:
i think that we need to validate the expects using something like `expect.assertions(1);`

or use instead

 `await expect(strategy.execute(payload)).rejects.toThrow(PaymentMethodCancelledError);`

this applies to all expect within a catch block